packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js

```diff
@@ -113,7 +113,19 @@ export class CheckoutPayments extends PureComponent {
     }
 
     renderPayments() {
-        const { paymentMethods } = this.props;
+        const { paymentMethods, selectPaymentMethod } = this.props;
+
+        if (paymentMethods.length === 1 && paymentMethods[0].code === 'free') {
+            const { code, title } = paymentMethods[0];
+
+            selectPaymentMethod({ code });
+
+            return (
+                <li block="CheckoutPayment">
+                    { title }
+                </li>
+            );
```

Comment:
Please call `renderPayment` with the single payment method item instead.

Reply:
Actually you can allow `return paymentMethods.map(this.renderPayment);` run.
As I understand, you just have to preselect the payment method here?

Reply:
The main difference is that paymentMethods draws items with a checkbox, and if use it, need to do the same check in paymentMethods, and still display just text

Is it better to do this check and other rendering in paymentMethods?

Reply:
I meant that even if we filter out other payment methods and have only one, it should have the same appearance as they would be more than one.
But I'm not a designer and may be wrong. @carinadues please correct if I'm wrong.

Reply:
Just in the description of the PR that Marita did, it is indicated that the checkbox needs to be removed(https://github.com/scandipwa/scandipwa/pull/4817#issuecomment-1144350185)